### ISSUE TITLE
Error executing luv callback: ...site/pack/packer/start/null-ls.nvim/lua/null-ls/loop.lua:108: handle 0x7f cba36eebe0 is already closing

### ISSUE THREAD

**Issue by kdheepak** [jose-elias-alvarez/null-ls.nvim#216]
### FAQ

- [X] I have checked [FAQ](https://github.com/jose-elias-alvarez/null-ls.nvim#faq) and it didn't resolve my problem.

### Issues

- [X] I have checked [existing issues](https://github.com/jose-elias-alvarez/null-ls.nvim/issues?q=is%3Aissue) and there are no issues with the same problem.

### Neovim Version

NVIM v0.6.0-dev+276-g685cf3981

### Steps to reproduce

Add the following lua configuration to neovim:

```lua
local null_ls = require("null-ls")
local helpers = require("null-ls.helpers")
local builtins = null_ls.builtins
local generator = null_ls.generator

local jet_julia = {
  method = null_ls.methods.DIAGNOSTICS,
  filetypes = { "julia" },
  -- null_ls.generator creates an async source
  -- that spawns the command with the given arguments and options
  generator = null_ls.generator({
    command = "jet",
    to_stdin = true,
    from_stderr = true,
    -- choose an output format (raw, json, or line)
    format = "json",
    check_exit_code = function(code)
      return code <= 1
    end,
    args = function(params)
      local filename = params.bufname
      return { filename }
    end,
    -- use helpers to parse the output from string matchers,
    -- or parse it manually with a function
    on_output = helpers.diagnostics.from_patterns({
      {
        pattern = [[:(%d+):(%d+) [%w-/]+ (.*)]],
        groups = { "row", "col", "message" },
      },
      {
        pattern = [[:(%d+) [%w-/]+ (.*)]],
        groups = { "row", "message" },
      },
    }),
  }),
}

null_ls.config({
  sources = {
    jet_julia,
  },
})
```

Create the following file in your environment path:

```julia
#!/usr/bin/env julia --project=~/.julia/environments/nvim-null-ls

using JET

filename = ARGS[1]

r = report_file(filename, analyze_from_definitions = true, annotate_types = true, toplevel_logger = stdout)

println(r)
```

Open the following file in neovim:

```julia
# demo.jl
# ====

# fibonacci
# ---------

fib(n) = n ≤ 2 ? n : fib(n-1) + fib(n-2)

fib(1000)   # never terminates in ordinal execution
fib(m)      # undef var
fib("1000") # obvious type error


# language features
# -----------------

# user-defined types, macros
struct Ty{T}
    fld::T
end

function foo(a)
    v = Ty(a)
    return bar(v)
end

# macros will be expanded
@inline bar(n::T)     where {T<:Number} = n < 0 ? zero(T) : one(T)
@inline bar(v::Ty{T}) where {T<:Number} = bar(v.fdl) # typo "fdl"
@inline bar(v::Ty)                      = bar(convert(Number, v.fld))

foo(1.2)
foo("1") # `String` can't be converted to `Number`

# even staged programming
# adapted from https://github.com/JuliaLang/julia/blob/9f665c19e076ab37cbca2d0cc99283b82e99c26f/base/namedtuple.jl#L253-L264
@generated function badmerge(a::NamedTuple{an}, b::NamedTuple{bn}) where {an, bn}
    names = Base.merge_names(an, bn)
    types = Base.merge_types(names, a, b)
    vals = Any[ :(getfield($(Base.sym_in(names[n], bn) ? :b : :a), $(names[n]))) for n in 1:length(names) ] # missing quote, just ends up with under vars
    :( NamedTuple{$names,$types}(($(vals...),)) )
end

badmerge((x=1,y=2), (y=3,z=1))
```

Running `jet demo.jl` from the command line works as expected:

<img width="1680" alt="Screen Shot 2021-09-29 at 1 17 24 PM" src="https://user-images.githubusercontent.com/1813121/135334193-40499dc2-6cf4-4d44-9c07-925df4a17a72.png">

### Expected behavior

I was expected it not to error. I want to develop a diagnostic for Julia and currently this appears to be blocking.

### Actual behavior

I get a `handle 0x7f cba36eebe0 is already closing` error.

### Debug log

```
$ cat ~/.cache/nvim/null-ls.log
cat: ~/.cache/nvim/null-ls.log: No such file or directory
```

### Help

Yes, but I don't know how to start. I would need guidance

**Comment by kdheepak** [comment]
I'm not getting this error anymore, maybe it had something to do with my script.

**Comment by jose-elias-alvarez** [comment]
I think there was an issue closing `stdin`, which should be fixed by f6da0ebf40fca161b776dad07b57d528a4b08a54. Let me know if you see any other issues!
